cmd/ably-boomer/main.go

```diff
@@ -0,0 +1,201 @@
+package main
+
+import (
+	"fmt"
+	"os"
+	"strings"
+
+	"github.com/ably-forks/boomer"
+	"github.com/ably/ably-boomer/perf"
+	"github.com/ably/ably-boomer/tasks/ably"
+	"github.com/inconshreveable/log15"
+	"github.com/urfave/cli/v2"
+)
+
+var log = log15.New()
+
+func nameToEnvVarNames(name, prefix string) []string {
+	envVar := strings.ToUpper(name)
+	envVar = strings.ReplaceAll(envVar, "-", "_")
+	envVar = fmt.Sprintf("%s_%s", prefix, envVar)
+	return []string{envVar}
+}
+
+func generateEnvVarNames(prefix string, flags []cli.Flag) {
+	for _, flag := range flags {
+		switch f := flag.(type) {
+		case *cli.BoolFlag:
+			f.EnvVars = nameToEnvVarNames(f.Name, prefix)
+		case *cli.DurationFlag:
+			f.EnvVars = nameToEnvVarNames(f.Name, prefix)
+		case *cli.Float64Flag:
+			f.EnvVars = nameToEnvVarNames(f.Name, prefix)
+		case *cli.Float64SliceFlag:
+			f.EnvVars = nameToEnvVarNames(f.Name, prefix)
+		case *cli.GenericFlag:
+			f.EnvVars = nameToEnvVarNames(f.Name, prefix)
+		case *cli.Int64Flag:
+			f.EnvVars = nameToEnvVarNames(f.Name, prefix)
+		case *cli.Int64SliceFlag:
+			f.EnvVars = nameToEnvVarNames(f.Name, prefix)
+		case *cli.IntFlag:
+			f.EnvVars = nameToEnvVarNames(f.Name, prefix)
+		case *cli.IntSliceFlag:
+			f.EnvVars = nameToEnvVarNames(f.Name, prefix)
+		case *cli.PathFlag:
+			f.EnvVars = nameToEnvVarNames(f.Name, prefix)
+		case *cli.StringFlag:
+			f.EnvVars = nameToEnvVarNames(f.Name, prefix)
+		case *cli.StringSliceFlag:
+			f.EnvVars = nameToEnvVarNames(f.Name, prefix)
+		case *cli.TimestampFlag:
+			f.EnvVars = nameToEnvVarNames(f.Name, prefix)
+		case *cli.Uint64Flag:
+			f.EnvVars = nameToEnvVarNames(f.Name, prefix)
+		case *cli.UintFlag:
+			f.EnvVars = nameToEnvVarNames(f.Name, prefix)
+		}
+	}
+}
+
+func taskFn(c *cli.Context) (func(), error) {
+	testType := c.String(testTypeFlag.Name)
+	switch testType {
+	case "fanout":
```

Comment:
if you're wanting to go the urfave route then fanout, personal, etc should probably be commands

Reply:
I would agree, but env vars seem to be easier to configure using our tooling: https://ably.atlassian.net/browse/REA-381?focusedCommentId=14391

Reply:
If commands would be nicer, could set the `CMD` to `ably-boomer $ABLY_TEST_TYPE` in the Docker image.

Reply:
Sounds good! I'll implement the tasks as subcommands then.

Reply:
Implemented in 0815c72.